fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1229,6 +1230,7 @@ public Builder setRunner(@Nullable FDBDatabaseRunner runner) {
          */
         public Builder setDatabase(@Nonnull FDBDatabase database) {
             this.runner = database.newRunner();
+            setRunnerDefaults();
```

Comment:
This is somewhat path dependent, but I think the online indexer used to use a database as its source of new contexts rather than the runner. (So, it kept an `FDBDatabase` around and would call `openContext` on it when it wanted a new transaction rather than a `FDBDatabaseRunner` from which it would call `openContext`.) I think it's actually still a reasonable way of specifying things: by saying that you want to run an index build against a certain database at this subspace (or path) and this meta-data, etc. The `setRecordStoreBuilder` method is then designed to give the user a quick way of specifying all of those things at once.

The `RecordStoreBuilder` that is kept by the `OnlineIndexer.Builder` is guaranteed to have a `null` context, and so it is theoretically agnostic to the database backing it. (So, for example, it will use the `Database` given to it at run time to evaluate its `KeySpacePath`.) That being said, it is *probably* an error if the user specifies a prototype record store or record store builder with a context against one database and then calls `setDatabase` with another. I think we could catch that by checking that none of the methods that update the `runner` ever change the database, and if they do, we can throw an error. Should I make that change?

Reply:
not as part of this change at least